src/caches/basic_method_caches.jl

```diff
@@ -116,3 +116,54 @@ function alg_cache(alg::RKMilCommute,prob,u,ΔW,ΔZ,p,rate_prototype,noise_rate_
   Kj = zero(u); Dgj = zero(noise_rate_prototype)
   RKMilCommuteCache(u,uprev,du1,du2,K,gtmp,L,WikJ,Dg,mil_correction,Kj,Dgj,tmp)
 end
+
+struct RKMil_GeneralConstantCache{WikType} <: StochasticDiffEqConstantCache
+  WikJ::WikType
+end
+
+@cache struct RKMil_GeneralCache{uType, rateType, rateNoiseType, WikType} <: StochasticDiffEqMutableCache
+  u::uType
+  uprev::uType
+  tmp::uType
+  du₁::rateType
+  du₂::rateType
+  K::uType
+  L::rateNoiseType
+  mil_correction::uType
+  ggprime::rateNoiseType
+  WikJ::WikType
+end
+
+function alg_cache(alg::RKMil_General,prob,u,ΔW,ΔZ,p,rate_prototype,noise_rate_prototype,uEltypeNoUnits,uBottomEltypeNoUnits,tTypeNoUnits,uprev,f,t,dt,::Type{Val{false}})
+  m = (typeof(ΔW) <: Number || is_diagonal_noise(prob))*1 + (alg.is_commutative)*2 + (!(typeof(ΔW) <: Number || is_diagonal_noise(prob) || alg.is_commutative))*3
+  WikJ = fill_WikJ(ΔW,Val(m),Val(false))
```

Comment:
@ChrisRackauckas I tried this. But I don't see how this might have any compile-time benefits?